toolkit/tools/pkg/imagecustomizerlib/bootcustomizer.go

```diff
@@ -160,37 +160,45 @@ func (b *BootCustomizer) UpdateKernelCommandLineArgs(defaultGrubFileVarName defa
 }
 
 // Makes changes to the /etc/default/grub file that are needed/useful for enabling verity.
-func (b *BootCustomizer) PrepareForVerity() error {
+func (b *BootCustomizer) PrepareForVerity(rootDeviceValue string) error {
 	if b.isGrubMkconfig {
 		// Force root command-line arg to be referenced by /dev path instead of by UUID.
-		defaultGrubFileContent, err := updateDefaultGrubFileVariable(b.defaultGrubFileContent, "GRUB_DISABLE_UUID",
+		defaultGrubFileContent, err := UpdateDefaultGrubFileVariable(b.defaultGrubFileContent, "GRUB_DISABLE_UUID",
 			"true")
 		if err != nil {
 			return err
 		}
 
 		// Disable recovery menu entry, to avoid having more than 1 linux command in the grub.cfg file.
 		// This will make it easier to modify the grub.cfg file to add the verity args.
-		defaultGrubFileContent, err = updateDefaultGrubFileVariable(defaultGrubFileContent, "GRUB_DISABLE_RECOVERY",
+		defaultGrubFileContent, err = UpdateDefaultGrubFileVariable(defaultGrubFileContent, "GRUB_DISABLE_RECOVERY",
 			"true")
 		if err != nil {
 			return err
 		}
 
+		// Check if rootDeviceValue is available as a field in BootCustomizer or another way
+		if rootDeviceValue != "" {
+			defaultGrubFileContent, err = UpdateDefaultGrubFileVariable(defaultGrubFileContent, "GRUB_DEVICE", rootDeviceValue)
+			if err != nil {
+				return err
+			}
+		}
+
 		b.defaultGrubFileContent = defaultGrubFileContent
 	}
 
 	return nil
 }
 
-func (b *BootCustomizer) WriteToFile(imageChroot *safechroot.Chroot) error {
+func (b *BootCustomizer) WriteToFile(imageChroot safechroot.ChrootInterface) error {
 	if b.isGrubMkconfig {
 		// Update /etc/defaukt/grub file.
-		err := writeDefaultGrubFile(b.defaultGrubFileContent, imageChroot)
+		fmt.Println("heheheheheheheh", b.defaultGrubFileContent)
```

Comment:
Remove.

(If it would be useful, you could add a trace log with `defaultGrubFileContent` contents.)